### PR TITLE
[TD]respect locale for date autofill

### PR DESCRIPTION
**Pull request by WandererFan**
This PR implements a fix for issue #18032.

### CODE REVIEW

**Comment by maxwxyz** [comment]
@WandererFan could you resolve the conflicts?

**Comment by maxwxyz** [comment]
@WandererFan ping for conflicts

**Comment by maxwxyz** [comment]
@WandererFan bump

File: src/Mod/TechDraw/Gui/DlgPrefsTechDrawAnnotationImp.cpp [hunk 1]
```diff
@@ -186,6 +188,8 @@ void DlgPrefsTechDrawAnnotationImp::loadSettings()
     loadLineStyleBoxes();
 
     ui->pcbBreakType->onRestore();
+
```
Comment (chennes) [inline src/Mod/TechDraw/Gui/DlgPrefsTechDrawAnnotationImp.cpp:191]:
```suggestion

```

File: src/Mod/TechDraw/Gui/DlgPrefsTechDrawAnnotationImp.cpp [hunk 2]
```diff
@@ -127,6 +127,8 @@ void DlgPrefsTechDrawAnnotationImp::saveSettings()
 
     ui->pcbBreakType->onSave();
     ui->pcbBreakStyle->onSave();
+
```
Comment (chennes) [inline src/Mod/TechDraw/Gui/DlgPrefsTechDrawAnnotationImp.cpp:130]:
```suggestion

```

**Comment by maxwxyz** [comment]
@WandererFan @chennes I get a build error on macOS while building #17564 for testing which seems related to this PR:
```
[1719/2669] Building CXX object src/Mod/TechDraw/App/CMakeFiles/TechDraw.dir/DrawTemplate.cpp.o
FAILED: src/Mod/TechDraw/App/CMakeFiles/TechDraw.dir/DrawTemplate.cpp.o 
/src/Mod/TechDraw/App/DrawTemplate.cpp:147:27: error: no member named 'setlocale' in namespace 'std'; did you mean simply 'setlocale'?
        auto timeLocale = std::setlocale(LC_TIME, nullptr);
                          ^~~~~~~~~~~~~~
                          setlocale
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/locale.h:53:8: note: 'setlocale' declared here
char            *setlocale(int, const char *);
                 ^
1 error generated.
ninja: build stopped: subcommand failed.
```

**Comment by maxwxyz** [comment]
@WandererFan  manually adding `#include <locale>` to `DrawTemplate.cpp`as mentioned by @kadet1090 solved the build issue

**Pull request merged**